File: bot/cogs/music/clear.py
import discord
from discord import app_commands
from discord.ext import commands

from ...handlers.music import MusicHandler
from ...handlers.queue import QueueHandler
from ...handlers.voice import VoiceHandler
from ...util.models import LavaBot


class Clear(commands.Cog):
    def __init__(self, bot: LavaBot) -> None:
        self.bot = bot
        self.music_handler = MusicHandler(bot)
        self.voice_handler = VoiceHandler(bot)
        self.queue_handler = QueueHandler(bot, self.voice_handler)

    @app_commands.command(description="Clears the whole (or part of the) queue")
    async def clear(self, interaction: discord.Interaction, index: int | None):

        if not index:
            await self.queue_handler.clear(interaction)
            return

        else:
            player = self.voice_handler.fetch_player(self.bot)
            if not player:
                return

            if index <= 0:
                await interaction.response.send_message(
                    ":warning: That index is too low! Queue starts at #1.",
                    ephemeral=True,
                )

            elif index > len(player.queue):

                await interaction.response.send_message(
                    f":warning: That index is too high! Queue only {len(player.queue)} items long.",
                    ephemeral=True,
                )

            else:
                await self.queue_handler.clear(interaction, index)

        return


async def setup(bot: LavaBot):
    await bot.add_cog(Clear(bot))

File: src/commands/music/clear.ts
import { SlashCommandBuilder } from "@discordjs/builders";
import { CommandInteraction } from "discord.js";
import { QueueHandler, VoiceHandler } from "../../util/handlers";
import { ExtendedClient, Command } from "../../util/structures";

export const command: Command = {
    data: new SlashCommandBuilder()
        .setName("clear")
        .setDescription("Clears the whole (or part of the) queue")
        .addNumberOption(option =>
            option
                .setName("index")
                .setDescription("Track you want to remove from the queue (check /list for those numbers)")
        )
        ,
    async execute(interaction: CommandInteraction, client: ExtendedClient) {
        const QH = new QueueHandler(client)
        const VH = new VoiceHandler(client)
        const player = await VH.ensureVoice(interaction)
        if (!player) {
            return
        }

        const index = interaction.options.getNumber('index')
        if (!index) {
            await QH.clear(interaction)
            return

        } else {
            if (index <= 0) {
                await interaction.reply({ content: ":warning: That index is too low! Queue starts at #1." , ephemeral: true})
                return

            } else if (index > player.queue.length) {
                await interaction.reply({content: `:warning: That index is too high! Queue only ${player.queue.length} items long`, ephemeral: true})
                return

            } else {
                await QH.clear(interaction, index)
                return
            }
        }
    }
}